analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Constraints/DisposableConstraint.cs

```diff
@@ -18,18 +18,15 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-using SonarAnalyzer.SymbolicExecution.Constraints;
+namespace SonarAnalyzer.SymbolicExecution.Constraints;
 
-namespace SonarAnalyzer.SymbolicExecution.Sonar.Constraints
+internal sealed class DisposableConstraint : SymbolicConstraint
 {
-    internal sealed class DisposableConstraint : SymbolicConstraint
-    {
-        public static readonly DisposableConstraint Disposed = new(ConstraintKind.DisposableDisposed);
-        public static readonly DisposableConstraint NotDisposed = new(ConstraintKind.DisposableNotDisposed);
+    public static readonly DisposableConstraint Disposed = new(ConstraintKind.DisposableDisposed);
+    public static readonly DisposableConstraint NotDisposed = new(ConstraintKind.DisposableNotDisposed);
```

Comment:
Try to delete this. It doesn't make sense to me to have it. Because nothing that was disposed cannot be un-disposed.
```suggestion
```

Reply:
From my understanding, this is there to be used only in the implementation of the abstract method `Opposite`.
I deleted the property and created a new Constrain directly in the Opposite method.

I'm not sure if this is really "cleaning" though.

Reply:
Return `null` from `Opposite` and delete `ConstraintKind.DisposableNotDisposed` too

Reply:
This is still missing

> and delete `ConstraintKind.DisposableNotDisposed` too